### PR TITLE
Add Test runs to CI, don't run `gh-pages` from forks

### PR DESCRIPTION
**Pull request by oldwomanjosiah**
Skipping the usual PR template, since this is just CI

### Changes

- Use matrix for builds, so that they can be done in parallel
- Add `test` workflow, using nextest to run all tests in repo
- Update formatting in all workflow files to follow yaml best practice
- Skip documentation workflow when dispatched on a fork

### CODE REVIEW

**Comment by oldwomanjosiah** [comment]
It might be worth it, to lower check times, to run the CI in a container image with the deps already installed. 

https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container

File: crates/eww/src/main.rs [hunk 1]
```diff
@@ -190,3 +190,13 @@ fn check_server_running(socket_path: impl AsRef<Path>) -> bool {
...
+
+
+
+
+
+
+
+
+
+
```
Comment (w-lfchen) [inline crates/eww/src/main.rs:202]: is this a feature? ^^

**Comment by oldwomanjosiah** [comment]
This is pretty old, and definitely outdated.

**Comment by w-lfchen** [comment]
i'll see whether i can pick up the ideas of this pr at some point if that's fine ^^

**Comment by oldwomanjosiah** [comment]
Yeah for sure! I don't run Linux as my daily right now so I can't really contribute there, but I'd love to see it get in fs

**Pull request closed without merging**